ivy/functional/ivy/nest.py

```diff
@@ -968,24 +1039,41 @@ def copy_nest(
     )
     if check_fn(nest, tuple):
         ret_list = [
-            copy_nest(i, include_derived=include_derived, to_mutable=to_mutable)
+            copy_nest(
+                i,
+                include_derived=include_derived,
+                to_mutable=to_mutable,
+                extra_nest_types=extra_nest_types,
+            )
             for i in nest
         ]
         if to_mutable:
             return ret_list
         return class_instance(tuple(ret_list))
-    elif check_fn(nest, list):
+    elif check_fn(nest, list) or isinstance(nest, extra_nest_types):
```

Comment:
isinstance(nest, (tuple, list) + extra_nest_types)

Reply:
This one is using `check_fn` for checking list type.